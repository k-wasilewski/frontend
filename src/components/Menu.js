import React, {Component} from 'react';
import '../css/App.css';
import {NavLink} from "react-router-dom";
import img from '../img/menu-btn.svg'
import {connect} from "react-redux";
import {setUsername} from "../redux/actions";

export class Menu extends Component {
    constructor(props) {
        super(props);

        this.menuRef = React.createRef();

        this.toggleMenuVisibility = this.toggleMenuVisibility.bind(this);
        this.logout = this.logout.bind(this);
    };

    toggleMenuVisibility() {
        this.menuRef.current.classList.toggle('hidden');
    };

    logout() {
        localStorage.removeItem('token');
        this.props.setUsername(null);
    }

    render() {
        return (
            <header>
                <div id="user">
                    {this.props.username}<br/>
                    {(this.props.username) ? <button onClick={this.logout}>Wyloguj</button> : null}
                </div>
                <div className="menu hidden" ref={this.menuRef}>
                    <nav>
                        <NavLink exact to={`/`} className="menu-item">
                            <button className='menu-item'>
                                <div className='menu-item-div'>Nowe zam√≥wienie</div>
                            </button>
                        </NavLink>
                        {this.props.username ? (
                            <NavLink to={`/summary`} className="menu-item">
                                <button className='menu-item'>
                                    <div className='menu-item-div'>Podsumowanie</div>
                                </button>
                            </NavLink>
                        ) : (
                            <NavLink to={`/login`} className="menu-item">
                                <button className='menu-item'>
                                    <div className='menu-item-div'>Logowanie</div>
                                </button>
                            </NavLink>
                        )}
                    </nav>
                </div>
                <img id='menu-btn' alt='menu button' onClick={this.toggleMenuVisibility} src={img} />
            </header>
        );
    };
}

function mapStateToProps(state) {
    return {
        username: state.setUsernameReducer.username
    };
}

const mapDispatchToProps = {
    setUsername
};

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
