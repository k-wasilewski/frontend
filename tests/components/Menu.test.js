import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../src/redux/store";
import {Provider} from "react-redux";
import Menu from "../../src/components/Menu";
import {BrowserRouter} from "react-router-dom";

describe("Menu specification", () => {
    it('renders two menu items and a menu btn', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <Menu />
                </BrowserRouter>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        let menuItems = tree.children[0]
        let newOrderBtn = menuItems.children[0].children[0]
        let summaryBtn = menuItems.children[0].children[1]
        expect(newOrderBtn.children[0].children[0].children[0]).toContain('Nowe zam√≥wienie')
        expect(summaryBtn.children[0].children[0].children[0]).toContain('Podsumowanie')

        let menuBtn = tree.children[1]
        expect(menuBtn.props.id).toContain('menu-btn')
    })
})