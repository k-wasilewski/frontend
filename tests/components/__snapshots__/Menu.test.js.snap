// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu specification renders two menu items and a menu btn 1`] = `
<header>
  <div
    className="menu hidden"
  >
    <nav>
      <a
        aria-current="page"
        className="menu-item active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <button
          className="menu-item"
        >
          <div
            className="menu-item-div"
          >
            Nowe zam√≥wienie
          </div>
        </button>
      </a>
      <a
        aria-current={null}
        className="menu-item"
        href="/summary"
        onClick={[Function]}
      >
        <button
          className="menu-item"
        >
          <div
            className="menu-item-div"
          >
            Podsumowanie
          </div>
        </button>
      </a>
    </nav>
  </div>
  <img
    alt="menu button"
    id="menu-btn"
    onClick={[Function]}
    src={Object {}}
  />
</header>
`;
